src/utilities/Dealer.cpp

```diff
@@ -0,0 +1,122 @@
+
+Dealer::DealerFlag::DealerFlag(DealerFlagTitle title, DealerFlagPriority priority)
+    : title(title), priority(priority) {}
+
+// Create a distribution of robots according to their flags
+std::unordered_map<std::string, v::RobotView> Dealer::distribute(const std::vector<v::RobotView>& allRobots, const FlagMap &flagMap) {
+    std::vector<std::vector<double>> scores = getScoreMatrix(allRobots, flagMap);
+    std::vector<int> assignment;
+
+    // solve the matrix and put the results in 'assignment'
+    rtt::Hungarian::Solve(scores, assignment);
+
+    return mapFromAssignments(allRobots, flagMap, assignment);
+}
+
+/* assignments now has the robot ids at the role index, and is ordered according to the roleNames
+* for example: assignments[0] = 2 // robot_id
+* and roleNames[0] = role_1
+* --> we can therefore make a map of <rolename, robot_id>
+*/
+std::unordered_map<std::string, v::RobotView> Dealer::mapFromAssignments(const std::vector<v::RobotView> &allRobots,
+                                                                  const Dealer::FlagMap &flagMap,
+                                                                  const std::vector<int> &assignment) const {
+    vector<string> orderedRoleNames;
+    for (auto const& [roleName, dealerFlags] : flagMap) {
+        orderedRoleNames.push_back(roleName);
+    }
+    unordered_map<string, v::RobotView> result;
+    for (int i = 0; i < orderedRoleNames.size(); i++) {
+        for (auto robot : allRobots) {
+            if (robot->getId() == assignment[i]) {
+                result.insert({orderedRoleNames[i], robot});
+            }
+        }
+    }
+    return result;
+}
+
+// Populate a matrix with scores
+std::vector<vector<double>> Dealer::getScoreMatrix(const std::vector<v::RobotView> &allRobots, const Dealer::FlagMap &flagMap) {
+    vector<vector<double>> scores;
+    scores.reserve(flagMap.size());
+    for (auto const& [roleName, dealerFlags] : flagMap) {
+        std::vector<double> row;
+        row.reserve(allRobots.size());
+        for (auto robot : allRobots) {
+            row.push_back(scoreForFlags(dealerFlags, robot));
+        }
+        scores.push_back(row);
+    }
+    return scores;
+}
+
+double Dealer::scoreForFlags(const std::vector<Dealer::DealerFlag> &dealerFlags, const v::RobotView &robot) {
+    double robotScore = 0;
+    for (auto flag : dealerFlags) {
+        robotScore += getScoreForFlag(robot, flag);
+    }
+    return robotScore;
+}
+
+double Dealer::getScoreForFlag(v::RobotView robot, Dealer::DealerFlag flag) {
+    double factor = getFactorForPriority(flag);
+    return factor * getDefaultFlagScores(robot, flag);
+}
+
+// TODO these values need to be tuned.
+double Dealer::getFactorForPriority(const Dealer::DealerFlag &flag) {
+    switch (flag.priority) {
+        case DealerFlagPriority::LOW_PRIORITY: return 1.0;
+        case DealerFlagPriority::MEDIUM_PRIORITY: return 2.0;
+        case DealerFlagPriority::HIGH_PRIORITY: return 3.0;
+    }
```

Comment:
return static_cast<int>(flag.priority) + 1.0;

Reply:
Personally think this is more explicit. Although your solution is less lines this is a lot easier to read.